src/pages/person-link/person.link.tsx

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable prettier/prettier */
+import { Link } from 'react-router-dom';
+import { PersonLinkProps } from '../../types/Person.link';
+
+export const PersonLink: React.FC<
+PersonLinkProps & { onClick?: () => void }
+> = ({ person, children, onClick }) => {
+  const handleClick = () => {
+    if (onClick) {
+      onClick();
+    }
+  };
+
+  const foundPerson = person ? person : null;
```

Comment:
You don't need _foundPerson_, you could do a conditional rendering as in the previous files:
```
return (
    <>
        {person ? (
           <Link ...
         ) : (
           <>{children}</>
    </>
)
```